internal/nexodus/wg_deploy.go

```diff
@@ -14,32 +14,37 @@ var (
 	securityGroupErr = errors.New("nftables setup error")
 )
 
-func (ax *Nexodus) DeployWireguardConfig(updatedPeers map[string]public.ModelsDevice) error {
+func (nx *Nexodus) DeployWireguardConfig(updatedPeers map[string]public.ModelsDevice) error {
 	cfg := &wgConfig{
-		Interface: ax.wgConfig.Interface,
-		Peers:     ax.wgConfig.Peers,
+		Interface: nx.wgConfig.Interface,
+		Peers:     nx.wgConfig.Peers,
 	}
 
-	if ax.TunnelIP != ax.getIPv4Iface(ax.tunnelIface).String() {
-		if err := ax.setupInterface(); err != nil {
+	if nx.TunnelIP != nx.getIPv4Iface(nx.tunnelIface).String() {
+		if err := nx.setupInterface(); err != nil {
 			return err
 		}
 	}
 
 	// add routes and tunnels for the new peers only according to the cache diff
+	var err error
 	for _, updatedPeer := range updatedPeers {
 		if updatedPeer.Id == "" {
 			continue
 		}
 		// add routes for each peer candidate (unless the key matches the local nodes key)
 		peer, ok := cfg.Peers[updatedPeer.PublicKey]
-		if !ok || peer.PublicKey == ax.wireguardPubKey {
+		if !ok || peer.PublicKey == nx.wireguardPubKey {
 			continue
 		}
-		ax.handlePeerRoute(peer)
-		ax.handlePeerTunnel(peer)
+		if err = nx.handlePeerRoute(peer); err != nil {
+			nx.logger.Errorf("Failed to handle peer route: %v", err)
+		}
+		if err = nx.handlePeerTunnel(peer); err != nil {
+			nx.logger.Errorf("Failed to handle peer tunnel: %v", err)
+		}
 	}
 
-	ax.logger.Debug("Peer setup complete")
-	return nil
+	nx.logger.Debug("Peer setup complete")
+	return err
```

Comment:
If there are two updated peers and the first fails, but the second succeeds, we won't return an error here. Maybe create a `firstError` variable that stores `err` the first time an error occurs and then return that here at the end.

I think we should return an error if any peer update fails and not just the last one.

Reply:
Ahh this was a dumb mistake 😵  You're right though, have fixed in latest commit

Reply:
not dumb. The error handling I'm suggesting is a little funky